packages/api-reference/src/hooks/useSidebar.ts

```diff
@@ -130,6 +131,30 @@ const items = computed(() => {
           }
         })
 
+  const groupOperations: SidebarEntry[] | undefined = parsedSpec.value?.[
+    'x-tagGroups'
+  ]
+    ? parsedSpec.value?.['x-tagGroups']?.map((tagGroup) => {
+        const children: SidebarEntry[] = []
+        tagGroup.tags.map((tagName: string) => {
+          const tag = operationEntries?.find(
+            (entry) => entry.title === tagName.toUpperCase(),
+          )
+          if (tag) {
+            children.push(tag)
+          }
+        })
+        const sidebarTagGroup = {
+          id: tagGroup.name,
+          title: tagGroup.name.toUpperCase(),
+          children,
+          show: true,
+          isGroup: true,
+        }
+        return sidebarTagGroup
+      })
```

Comment:
This business

Reply:
@amritk imma be honest, i was cooking this at 1am and i had no idea what i was doing in es6 land 😭

Reply:
Alright once I got the spec, this code isn't the worst. BUT I will throw up a ticket for perf improvements on `useSidebar` as a whole as we can definitely combine some loops